modules/users/server/models/user.server.model.js

```diff
@@ -147,4 +278,36 @@ UserSchema.statics.findUniqueUsername = function(username, suffix, callback) {
 	});
 };
 
-mongoose.model('User', UserSchema);
+UserSchema.statics.oAuthHandle = function (currentUser, provider, id, accessToken, refreshToken, providerData, userData, callback) {
+	if (!currentUser) {
+		UserModel.findByProvider(provider, id, function (err, user) {
+			if (err) {
+				return callback(err);
+			}
+			if (!user) {
+				var possibleUsername = userData.username || ((userData.email) ? userData.email.split('@')[0] : '');
+				return UserModel.findUniqueUsername(possibleUsername, null, function (availableUserName) {
+					user = new UserModel(userData);
+					user.setIdentity(provider, id, accessToken, refreshToken, providerData);
+					user.setIdentity(UserModel.USERNAME, availableUserName, null, null, null);
+					user.save(function (err) {
+						return callback(err, user, true);
+					});
+				});
+			}
```

Comment:
will need to add here `user.setIdentity(provider, id, accessToken, refreshToken, providerData);` to allow updates